src/runtime_src/driver/xclng/drm/xocl/userpf/xocl_ioctl.c

```diff
@@ -214,11 +220,6 @@ int xocl_ctx_ioctl(struct drm_device *dev, void *data,
 		// CU unlocked explicitly
 		--client->num_cus;
 		--xdev->ip_reference[args->cu_index];
-		if (!client->num_cus) {
```

Comment:
Why not unlock here if you removed the last context?

Reply:
yes